src/attr/__init__.pyi

```diff
@@ -223,7 +223,7 @@ def make_class(
 def asdict(
     inst: Any,
     recurse: bool = ...,
-    filter: Optional[_FilterType] = ...,
+    filter: Optional[_FilterType[_T]] = ...,
```

Comment:
I don't think there is any point to adding a TypeVar here if it is only used in one place within the signature.  The idea is to create a relationship to another argument, another method on the class (not applicable here), or the result.